aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,296 @@
+
+.. code-block:: c#
+
+    .UseContentRoot("c:\\mywebsite")
+
+Detailed Errors ``bool``
+    Key: ``detailedErrors``. Defaults to ``false``. When ``true`` (or when Environment is set to "Development"), the app will display details of startup exceptions, instead of just a generic error page. Set using ``UseSetting``.
+
+.. code-block:: c#
+
+    .UseSetting("detailedErrors", "true")
+
+When Detailed Errors is set to ``false`` and Capture Startup Errors is ``true``, a generic error page is displayed in response to every request to the server.
+
+.. image:: hosting/_static/generic-error-page.png
+
+When Detailed Errors is set to ``true`` and Capture Startup Errors is ``true``, a detailed error page is displayed in response to every request to the server.
+
+.. image:: hosting/_static/detailed-error-page.png
+
+Environment ``string``
+    Key: ``environment``. Defaults to "Production". May be set to any value. Predefined values include "Development" and "Staging" in addition to "Production". Values are not case sensitive. See :doc:`environments`. Set using the ``UseEnvironment`` method.
+
+.. code-block:: c#
+
+    .UseEnvironment("Development")
+
+.. note:: By default, the environment is read from the ``ASPNETCORE_ENVIRONMENT`` environment variable. When using Visual Studio, the environment will be based on settings in the *launchSettings.json* file.
+
+Server URLs ``string``
+    Key: ``urls``. Set to a semicolon (;) separated list of URL prefixes to which the server should response. For example, "http://localhost:123". The domain/host name can be replaced with "*" to indicate the server listen to any request coming to any IP address or host on the specified port (for example, "http://*:5000"). The protocol ("http://" or "https://") must be included with each URL.
+
+.. code-block:: c#
+
+    .UseUrls("http://*:5000;http://localhost:5001;https://hostname:5002")
+
+Startup Assembly ``string``
+    Key: ``startupAssembly``. Determines the assembly to search for the ``Startup`` class (unless specified using ``WebHostBuilder.UseStartup<StartupType>``, which overrides this setting). Set using non-generic ``UseStartup`` method.
+
+.. code-block:: c#
+
+    .UseStartup("StartupAssemblyName")
+
+Web Root ``string``
+    Key: ``webroot``. If not specified, Web Root will use the Content Root path. Determines default root location for static files to be served by the app (relative to Content Root). Set using ``UseWebRoot``.
+
+.. code-block:: c#
+
+    .UseWebRoot("public")
+
+A host can use :doc:`configuration` to set its values, which may be subsequently overridden. This is specified using the ``UseConfiguration`` method.
```

Comment:
Use :doc:`configuration` to set configuration values. The vaules may be subsequently overridden (for example, by ...). Specify configuration values with ``UseConfiguration``.